polly/src/event_manager.rs

```diff
@@ -0,0 +1,452 @@
+            }
+            None => {
+                println!("Pollable id {} not found", pollable);
+                return Err(Error::PollableNotFound(pollable));
+            }
+        }
+        Ok(())
+    }
+
+    fn dispatch_event(
+        &mut self,
+        source: Pollable,
+        evset: epoll::Events,
+        wrapped_handler: WrappedHandler,
+    ) -> Result<()> {
+        let mut all_ops = Vec::new();
+
+        // If an error occurs on a fd then only dispatch the error callback,
+        // ignoring other flags.
+        if evset.contains(epoll::Events::EPOLLERR) {
+            if let Some(mut ops) = wrapped_handler
+                .lock()
+                .expect("Handler lock is poisoned")
+                .handle_error(source.clone())
+            {
+                all_ops.append(&mut ops);
+            }
+        } else {
+            if evset.contains(epoll::Events::EPOLLIN) {
+                if let Some(mut ops) = wrapped_handler
+                    .lock()
+                    .expect("Handler lock is poisoned")
+                    .handle_read(source.clone())
+                {
+                    all_ops.append(&mut ops);
+                }
+            }
+            if evset.contains(epoll::Events::EPOLLOUT) {
+                if let Some(mut ops) = wrapped_handler
+                    .lock()
+                    .expect("Handler lock is poisoned")
+                    .handle_write(source.clone())
+                {
+                    all_ops.append(&mut ops);
+                }
+            }
+            if evset.contains(epoll::Events::EPOLLRDHUP) {
+                if let Some(mut ops) = wrapped_handler
+                    .lock()
+                    .expect("Handler lock is poisoned")
+                    .handle_close(source.clone())
+                {
+                    all_ops.append(&mut ops);
+                }
+            }
+        }
+
+        self.run_pollable_ops(wrapped_handler.clone(), all_ops)?;
+        Ok(())
+    }
+
+    fn process_events(&mut self, event_count: usize) -> Result<()> {
+        for idx in 0..event_count {
+            let event = self.events[idx];
+            let event_mask = event.events;
+            let event_data = event.data;
+            let evset = match Events::from_bits(event_mask) {
+                Some(evset) => evset,
+                None => {
+                    println!("epoll: ignoring unknown event set: 0x{:x}", event_mask);
+                    continue;
+                }
+            };
+
+            if let Some(event_handler_data) = self.handlers.get(&(event_data as i32)) {
+                self.dispatch_event(
+                    event_handler_data.data.0.clone(),
+                    evset,
+                    event_handler_data.handler,
+                )?;
+            }
+        }
+        Ok(())
+    }
+
+    // Wait for events, then dispatch to registered event handlers.
+    pub fn run(&mut self) -> Result<usize> {
```

Comment:
I understand this possibility, but this can be handled by actually implementing the eventhandler trait for some other object owned by the eventmanager owner. 

I am not sure why would we prefer to pass this concern to another component.